src/includes/class-health-check-wp-debug.php

```diff
@@ -0,0 +1,536 @@
+			'message' => esc_html__( 'WP_DEBUG_DISPLAY was disabled.', 'health-check' ),
+		);
+
+		wp_send_json_success( $response );
+
+	}
+
+	/**
+	 * Check if wp-config backup exists
+	 *
+	 * @uses file_exists()
+	 *
+	 * @return bool true/false depending if the backup exists
+	 */
+	static function check_wp_config_backup() {
+		$wpconfig_backup = ABSPATH . 'wp-config_hc_backup.php';
+
+		if ( file_exists( $wpconfig_backup ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Creates a backup of wp-config.php
+	 *
+	 * @uses file_exists()
+	 * @uses copy()
+	 *
+	 * @return string $output Success/Fail
+	 */
+	static function create_wp_config_backup() {
+		$wpconfig        = ABSPATH . 'wp-config.php';
+		$wpconfig_backup = ABSPATH . 'wp-config_hc_backup.php';
+
+		if ( ! copy( $wpconfig, $wpconfig_backup ) ) {
+			$response = array(
+				'status'  => 'error',
+				'message' => esc_html__( 'wp-config.php backup failed.', 'health-check' ),
+			);
+			wp_send_json_error( $response );
+		}
+
+		$response = array(
+			'status'  => 'success',
+			'message' => esc_html__( 'wp-config.php backup was created.', 'health-check' ),
+		);
+
+		wp_send_json_success( $response );
+	}
+
+	/**
+	 * Restores a backup of wp-config.php
+	 *
+	 * @uses copy()
+	 * @uses unlink()
+	 *
+	 * @return string $output Success/Fail
+	 */
+	static function restore_wp_config_backup() {
+		$wpconfig        = ABSPATH . 'wp-config.php';
+		$wpconfig_backup = ABSPATH . 'wp-config_hc_backup.php';
+
+		if ( ! copy( $wpconfig_backup, $wpconfig ) ) {
+			$response = array(
+				'status'  => 'error',
+				'message' => esc_html__( 'wp-config.php restore failed.', 'health-check' ),
+			);
+			wp_send_json_error( $response );
+		}
+
+		$response = array(
+			'status'  => 'success',
+			'message' => esc_html__( 'wp-config.php backup was restored.', 'health-check' ),
+		);
+
+		unlink( $wpconfig_backup );
+
+		wp_send_json_success( $response );
+	}
+
+	/**
+	 * Restores a backup of wp-config.php on plugin deactivation
+	 *
+	 * @uses file_exists()
+	 * @uses copy()
+	 * @uses unlink()
+	 *
+	 * @return void
+	 */
+	static function deactivation_restore_wp_config_backup() {
+		$wpconfig        = ABSPATH . 'wp-config.php';
+		$wpconfig_backup = ABSPATH . 'wp-config_hc_backup.php';
+
+		if ( file_exists( $wpconfig_backup ) ) {
+			if ( copy( $wpconfig_backup, $wpconfig ) ) {
+				unlink( $wpconfig_backup );
+			}
+		}
+
+		return;
+	}
+
+	/**
+	 * Read debug.log contents
+	 *
+	 * @uses file_exists()
+	 * @uses fopen()
+	 * @uses die()
+	 * @uses fwrite()
+	 * @uses fclose()
+	 * @uses WP_CONTENT_DIR
+	 * @uses wp_die()
+	 * @uses wp_send_json_success()
+	 *
+	 * @return void
+	 */
+	static function read_wp_debug() {
+
+		// check if debug.log exists else create it to avoid error.
+		if ( ! file_exists( WP_CONTENT_DIR . '/debug.log' ) ) {
+			$debug_log = fopen( WP_CONTENT_DIR . '/debug.log', 'w' ) or die( 'Cannot create debug.log!' );
+			fwrite( $debug_log, '' );
```

Comment:
We should avoid truncating the file on every read, if something happens and the user leaves the page they've now lost their logs. Truncate when disabling logging instead.